### DISCUSSION TITLE
Help not showing in ps 1.7.8 with classic theme

### DISCUSSION THREAD

**Discussion by caciquemar** [Pixel-Open/prestashop-cloudflare-turnstile#10] [Q&A]
HI, 
I ve installed the module, 
but its not showing at register form

for the other forms i've added the following code at the correspondent tpl file :
{widget name='pixel_cloudflare_turnstile' custom='1' action='custom-form'}

but fail to show the turnstile validator

Thanks

**Comment by magentix** [comment]
Hi,

The following widget with options:

`{widget name='pixel_cloudflare_turnstile' custom='1' action='custom-form'}`

is only for custom forms.

First you need to configure the module and select forms you want to protect. Then manually add the widget in templates:

`{widget name='pixel_cloudflare_turnstile'}`

- Contact: themes/classic/modules/contactform/views/templates/widget/contactform.tpl
- Login: themes/classic/templates/customer/_partials/login-form.tpl
- Reset password: themes/classic/templates/customer/password-email.tpl

**Comment by caciquemar** [comment]
Thanks for your answer

done what you have suggested 
but still not showing


this is how i added the widget in the password email template:

    <section class="form-fields">
     
      <div class="form-group center-email-fields">
      <div> {widget name='pixel_cloudflare_turnstile'} </did>
        <label class="col-md-3 form-control-label required">{l s='Email address' d='Shop.Forms.Labels'}</label>
        <div class="col-md-5 email">
          <input type="email" name="email" id="email" value="{if isset($smarty.post.email)}{$smarty.post.email|stripslashes}{/if}" class="form-control" required>
        </div>
        
        <button class="form-control-submit btn btn-primary hidden-xs-down" name="submit" type="submit">
          {l s='Send reset link' d='Shop.Theme.Actions'}
        </button>
        <button class="form-control-submit btn btn-primary hidden-sm-up" name="submit" type="submit">
          {l s='Send' d='Shop.Theme.Actions' }
         
        </button>
       
      
      </div>
    </section>

when i load the page i got this:

<div class="pxd-cloudflare-turnstile">
    <div class="cf-turnstile" data-sitekey="j=here goes site key provided by cluoudfare" data-theme="auto" data-action="password"></div>

but it does not show 
im attaching a print screen


![Captura de pantalla (204)](https://github.com/Pixel-Open/prestashop-cloudflare-turnstile/assets/7314166/555d086f-b720-48d3-b4b4-4de778f7049f)

**Comment by magentix** [comment]
Ok the widget has been added in the page.

Have you this script in the head?

`<script type="text/javascript" src="https://challenges.cloudflare.com/turnstile/v0/api.js" async></script>`

**Comment by caciquemar** [comment]
No, I dont see that script...

**Comment by caciquemar** [comment]
sorry but why i dont have that scrip ?

**Comment by magentix** [comment]
Hello,

We use the hook "ActionFrontControllerSetMedia" to add the Turnstile script. What is the version of your Prestashop? Does this hook exist in your Prestashop? (search for "Hook::exec('actionFrontControllerSetMedia', []);" in all project).

Can you try to reinstall the module to register hook again?

**Comment by caciquemar** [comment]
![Screenshot](https://github.com/Pixel-Open/prestashop-cloudflare-turnstile/assets/7314166/c7ffd34c-77fd-4941-ba01-0b4f17829e35)


thanks for your answer
I do have the hook acctionfrontcontrollersetmedia and its set to the module (im attaching a print screen)

we are using presta 1.7.8.10
